Update README.md, add documentation

maria8ch opened the pull request:
This is for Issue #4

maria8ch committed:
Update README.md

maria8ch committed:
Update index.md

maria8ch committed:
Update index.md

maria8ch committed:
Update index.md

Mostly formatting, some clarifications. Not sure where current Line 42 should be placed properly, feels incomplete. Last time should be the note that's currently LIne 41.

maria8ch committed:
Update content index.md

I changed one label because it wasn't accurate.

maria8ch commented:
Almost done with the page, just need someone to look over it and help clarify certain steps.

maria8ch committed:
Update index.md

maria8ch commented:
Finished Issue #4 . Need reviews and a merge.

darrenhunt2 commented on docs/index.md:
```diff
@@ -1,3 +1,45 @@
+* Generate plots to monitor the output of the above.
+
+
+My suggestion of a place to start would be to find a dozen or so images from a focus sequence at MRO, some above, some near, and some below focus. This will give us some test data to play with. We would also want the logbooks that go with those images, so that we can compare with the focus values measured by the observers.
```
Maybe reword this line:

"Starting with 10 images at, above, and below focus...."

darrenhunt2 commented on docs/index.md:
```diff
@@ -1,3 +1,45 @@
+
+My suggestion of a place to start would be to find a dozen or so images from a focus sequence at MRO, some above, some near, and some below focus. This will give us some test data to play with. We would also want the logbooks that go with those images, so that we can compare with the focus values measured by the observers.
+
+I would next try to use the astropy affiliate photutils package to detect and measure point sources on each image and compute the mean full-width at half maximum (FWHM) for that image. One thing I don't know is how well photutils does on images that are fairly far from correct focus, such that the stars look more like donuts than points. This would be something to explore.
```
Maybe: "Next, we will..."

darrenhunt2 commented on docs/index.md:
```diff
@@ -1,3 +1,45 @@
+
+Two ways: Measure FWHM and take the smallest value; look for the image the star is the brightest (small radius)
+
+Note: the FWHM method does not work for “donuts” -- extremely out of focus stars.
```
The rest of this seems to look good/is pretty self explanatory. I think this will work otherwise.

darrenhunt2 reviewed (changes requested):
Seems good other than the two comments that I left

parejkoj commented:
Don't worry, I'll get to my review...

maria8ch committed:
Update index.md

parejkoj commented on docs/index.md:
```diff
@@ -1,3 +1,43 @@
+## Detailed explanation
+
+### Focus math
+* Starting with 10 images at, above, and below focus (taken from MRO with logbooks intact), with some above, some near, and some below focus. This will give us some test data to play with. Logbooks are necessary to compare with the focus values measured by the observers.
```
The first sentence has duplication of the "above, near, below focus" bit.

parejkoj commented on docs/index.md:
```diff
@@ -1,3 +1,43 @@
+
+* Use the astropy affiliate photutils package to detect and measure point sources on each image, compute the mean full-width at half maximum (FWHM) for that image.
+
+* Write the code to do the focusing. This will be done before the automating part (that communicates with the telescope).
```
"do the focusing": did you mean "compute the optimum focus from the set of images"?

parejkoj commented on docs/index.md:
```diff
@@ -1,3 +1,43 @@
+
+Our program needs to: 
+  * Pick a bright star in image, measure its FWHM (keep in mind that width of Gaussian is not the same as FWHM)
+  * Inside, outside (above, below) focus
```
I think you meant here "take images at a range of different focuser positions"

maria8ch replied:
i changed some wording. is this good or should i just delete the two lines and replace with your quotation mark words?

parejkoj commented on docs/index.md:
```diff
@@ -1,3 +1,43 @@
-These pages will describe the automated focusing system being developed by students in UW Astronomy Undergrad Engineering Group (AUEG).
+These pages will describe the automated focusing system being developed by students in UW Astronomy Undergrad Engineering Group (AUEG). We are writing in python. There are two ways to achieve focusing:  
+a) measure FWHM and take the smallest value  
+b) look for the brightest pixels in the image to calculate radius of star and go from there
```
The "brightest pixel" method does not require that you "calculate radius of star". It's based entirely on getting as much light focused onto a single pixel as possible. If you're going to mention it here, it would be good to also mention it below as something to check in the focus images, as a comparison with the star FWHM.

parejkoj commented on docs/index.md:
```diff
@@ -1,3 +1,43 @@
+a) measure FWHM and take the smallest value  
+b) look for the brightest pixels in the image to calculate radius of star and go from there
+
+## Here is a list of steps to take, in order:
```
Shorter header please. Maybe something like: `## Steps to build automated focuser`?

parejkoj reviewed (changes requested):
Thanks for collating the notes here.

Overall comment: If you put each sentence on its own line, it is easier to follow changes to those sentences. Markdown automatically merges sentences into paragraphs, so long as there is not an empty line between them.

maria8ch reviewed (commented):
Replied to John's comments, changed some stuff. Are we good now?

maria8ch committed:
Resolve John's comments

Changed some stuff according to John's review- here we are. If this is final then we should be able to merge things. If not, only minor tweaks are left.

parejkoj reviewed (approved):
Only other comment is "focus so that those pixels saturate": it's not so they saturate, just so that there is as much flux as possible in the brightest pixels.

maria8ch committed:
Update index.md

Final update and we are good to go for a merge :)

maria8ch commented:
ok so just so i'm clear- I changed the last little thing on the doc, so now I can click the green merge pull request button, right?

parejkoj commented:
Your changes are approved by both reviewers: merge away!

And then we can see if the changes are propagated to the website...
